ambuild2/frontend/cpp/compilers.py

```diff
@@ -59,7 +59,7 @@ def __init__(self, options = None):
   def inherit(self, other):
     self.debuginfo = other.debuginfo
     for attr in self.attrs:
-      setattr(self, attr, copy.copy(getattr(other, attr)))
+      setattr(self, attr, copy.deepcopy(getattr(other, attr)))
```

Comment:
Good catch, though this is a big enough change to definitely warrant its own commit.